### PR TITLE
Fixes heretic summon rituals not consuming items

### LINKED ISSUE #85925: heretic's rust walker ritual producing infinite rust walkers without consuming materials
## Issue Summary

rust heretic did the ritual to summon a rust walker, the materials weren't consumed, millions must rust

## Reproduction:

1) be rust heretic
2) make rust walker
3) it seemingly does not consume the materials
4) make 25 additional rust walkers

the only seemingly unusual part about this is that the rune was in space but otherwise im not quite sure what caused it (all the bodies were dragged there by the 25 rust walkers)

image

notably theres a dupe issue ([#51717](https://github.com/tgstation/tgstation/issues/51717)) but its like four years old so

### PR DESCRIPTION
**Pull request by Metekillot**
## About The Pull Request

Heretic summon rituals are broken right now to not consume items; this means infinite summons for one set of items. This PR fixes that.
## Why It's Good For The Game
Exploits bad
Fixes https://github.com/tgstation/tgstation/issues/85925
## Changelog
:cl: Bisar
fix: The Nanotrasen Emergency Religious Response Team has blessed the station; heretic summon rituals now actually consume resources, fixing infinite summon loops.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/heretic/heretic_knowledge.dm [hunk 1]
```diff
@@ -540,7 +540,8 @@
 	var/mob/living/mob_to_summon
 
 /datum/heretic_knowledge/summon/on_finished_recipe(mob/living/user, list/selected_atoms, turf/loc)
-	summon_ritual_mob(user, loc, mob_to_summon)
+	. = summon_ritual_mob(user, loc, mob_to_summon)
+	return .
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/heretic_knowledge.dm:544]:
```suggestion
	return summon_ritual_mob(user, loc, mob_to_summon)
```

File: code/modules/antagonists/heretic/knowledge/side_ash_moon.dm [hunk 2]
```diff
@@ -84,10 +84,4 @@
 	depth = 10
 
 /datum/heretic_knowledge/summon/ashy/cleanup_atoms(list/selected_atoms)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/knowledge/side_ash_moon.dm:86]: There's nothing in this proc override anymore, just delete it.

File: code/modules/antagonists/heretic/knowledge/side_rust_cosmos.dm [hunk 3]
```diff
@@ -105,10 +105,4 @@
 	depth = 8
 
 /datum/heretic_knowledge/summon/rusty/cleanup_atoms(list/selected_atoms)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/knowledge/side_rust_cosmos.dm:107]: Same here

**Review by MrMelbert — COMMENTED** [review]
Maybe we should assert `on_finished_recipe` returns `TRUE` or `FALSE`, not `null`? Would make this easier to catch in the future

**Comment by Xkallubar** [comment]
this is so sad, infinite free antag generator glitch is getting patched

**Pull request merged**